flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/PendingCheckpointTest.java

```diff
@@ -578,6 +580,55 @@ public void testReportTaskFinishedOperators() throws IOException {
                 contains(ACK_TASKS.get(0).getVertex()));
     }
 
+    /**
+     * We need to trigger disposal prior completing the checkpoint future to avoid a race condition
+     * with the CheckpointCleaner shutdown.
+     */
+    @Test
+    public void testAbortTriggersDisposalPriorCompletingCheckpointFuture() throws Exception {
+        final PendingCheckpoint pending =
+                createPendingCheckpoint(
+                        CheckpointProperties.forCheckpoint(
+                                CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION));
+        final CompletableFuture<Void> completionFuture =
+                pending.getCompletionFuture()
+                        .handle(
+                                (result, error) -> {
+                                    assertNull(result);
+                                    assertNotNull(error);
+                                    assertTrue(pending.isDisposed());
+                                    return null;
+                                });
```

Comment:
```suggestion
                        .thenAccept(
                                result -> assertTrue(pending.isDisposed()));
```

If the future fails we will fail later on anyway, so we can focus on whether the checkpoint is disposed or not for the assertion.

Reply:
👍

Reply:
This one is already completed exceptionally, so we actually need a handle here. We're basically asserting that it has been completed exceptionally and checkpoint has been disposed. For the 2nd case we can do what you're suggesting